### DISCUSSION TITLE
Is supabase supported?

### DISCUSSION THREAD

**Discussion by nash1111** [diesel-rs/diesel#3448] [Q&A] [answered]
I try to connect using supabase connection string but it fails as follows

```
cargo new stmterror
cd stmterror
# add Cargo.toml
diesel = { version = "2.0.0", features = ["postgres"] }
dotenvy = "0.15"
# edit .env
DATABASE_URL="supabase connection string"
# and run
diesel setup
diesel migration run
# fails
prepared statement "__diesel_stmt_0"
already exists
```
I have also tried the following
```
DATABASE_URL="supabase connection string with wrong password"
# fails
FATAL:  SASL authentication failed

DATABASE_URL="connection string of psql that i built with docker"
# succeeded without a message
```

**Comment by weiznich** [comment, marked as answer]
Diesel does officially support postgresql, mysql and sqlite. Any other database system is not supported by diesel itself. If the database system supports connecting via one of the native client libraries it might work with diesel, but we neither guarantee that nor do we consider it as a bug if certain features do not work as expected. We generally advice people to write a custom third party backend implementation for such systems, because that gives them the required control to make that backend behave as required by the underlying database system.

In this specific case the error message suggests that the database system does not support prepared statements, which is something that is fundamental to all diesel provided connection implementations.

**Comment by nash1111** [comment]
Thanks for the reply.
I will use postgresql without supabase.
